materialize-pinecone/client/client.go

```diff
@@ -80,6 +87,17 @@ func (c *OpenAiClient) CreateEmbeddings(ctx context.Context, input []string) ([]
 	}
 	defer res.Body.Close()
 
+	if res.StatusCode == http.StatusTooManyRequests {
+		var errorBody openAiEmbeddingsError
+		if err := json.NewDecoder(res.Body).Decode(&errorBody); err != nil {
+			log.WithField("error", err).Warn("could not decode error response body")
+		} else if errorBody.Error.Message != "" {
+			return nil, fmt.Errorf("error creating embeddings: %s", errorBody.Error.Message)
```

Comment:
should we include the status code in the log as well? might be useful

Reply:
That does seem like it could be helpful - added the status to the error message.